src/VarDriver.cpp

```diff
@@ -1943,6 +1943,134 @@ Projection::ProjectionType VarDriver::projectionFromConfig()
   return retVal;
 }
 
+// This routine from the thermo branch, NetCDF_XYZ::readNetCDF
+
+int VarDriver::readNetCDF_thermo(const char* filename){
```

Comment:
this function seems similar to the function from https://github.com/mmbell/samurai/blob/main/src/thermo_reference/NetCDF_XYZ.cpp#L70. Can we modify that existing function instead of adding a new one here?

Reply:
While it is similar it reads different variables.  For the time being, we would like to perform a direct transplant of code from the thermo branch into the main branch to minimize the likelihood that we introduce any errors.  Once a functional version of the Thermo capability, we will have a separate PR that cleans up duplicate code.

Reply:
Hi Jian! Yes, that's where this function is mostly copied from. Our strategy for porting the thermo code is to

1.  take the few functions we need from the thermo_reference code in the 'thermo' branch and integrate them into the samurai_wind code. The code from thermo_reference needs to be minorly updated and cleaned of QT calls so it runs
2. Keep doing this until we can run and produce correct thermo results 
3. Once we're producing correct results, then eliminate redundancies in functions and begin optimization work. (This function will eventually be folded into the class_nc() reader I think)

Reply:
Thanks for the clarification. That is clear to me now.